build.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::process::Command;
+
+// this build script is called on en every `devtool build`,
+// embedding the FIRECRACKER_VERSION directly in the resulting binary
+fn main() {
+    let firecracker_version = Command::new("bash")
+        // the dirty flag reported by git describe --dirty does not suffice,
+        // since it is not affected by new, unstaged files;
+        // we check the output of git status to account for this case.
+        .args(&[
+            "-c",
+            "err=0; trap \"err=1\" ERR; git describe >/dev/null;\
+            echo -n \"$(git describe)\";\
+            [[ $(git status --porcelain | wc -l) -eq 0 ]] && : || echo '-dirty';\
+            test $err = 0",
+        ])
```

Comment:
This looks complicated enough that I actually prefer just the simple `git describe --dirty` even if it misses untracked files.
From my point of view, this mechanism is to help us avoid _accidentally_ confusing a dirty binary with a release one. _Accidentally_ having untracked files that cleanly compile is highly improbable and I'm not worried about covering it.

The above is just a preference, I am also fine with the current "complicated" version.

Reply:
I am also in favor of the less complicated output (i.e "git describe only").

Reply:
It seems that this is the general opinion, so I will roll back to `git describe --version`